client/landing/stepper/declarative-flow/internals/steps-repository/use-my-domain/index.tsx

```diff
@@ -11,32 +11,56 @@ import {
 	UseMyDomainInputMode,
 } from 'calypso/components/domains/connect-domain-step/constants';
 import UseMyDomainComponent from 'calypso/components/domains/use-my-domain';
+import { useSiteData } from 'calypso/landing/stepper/hooks/use-site-data';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
 import { domainMapping, domainTransfer } from 'calypso/lib/cart-values/cart-items';
 import CalypsoShoppingCartProvider from 'calypso/my-sites/checkout/calypso-shopping-cart-provider';
+import { useDispatch } from 'calypso/state';
+import { setSelectedSiteId } from 'calypso/state/ui/actions';
 import { shouldUseStepContainerV2 } from '../../../helpers/should-use-step-container-v2';
 import type { Step as StepType } from '../../types';
 
 import './style.scss';
 
+type OwnershipVerificationData = {
+	ownership_verification_data: {
+		verification_type: 'auth_code';
+		verification_data: string;
+	};
+};
+
 const UseMyDomain: StepType< {
 	submits:
 		| {
 				mode: 'transfer' | 'connect';
 				domain: string;
 		  }
-		| { domainCartItem: MinimalRequestCartProduct }
+		| {
+				domainCartItem: MinimalRequestCartProduct;
+				verificationData?: OwnershipVerificationData;
+				onDone?: ( error?: Error ) => void;
+		  }
 		| undefined;
 } > = function UseMyDomain( { navigation, flow } ) {
 	const { __ } = useI18n();
 	const { goNext, goBack, submit } = navigation;
 	const location = useLocation();
 	const isDomainConnectionRedesign = config.isEnabled( 'domain-connection-redesign' );
+	const reduxDispatch = useDispatch();
+	const { siteId } = useSiteData();
 
 	const [ useMyDomainMode, setUseMyDomainMode ] = useState< UseMyDomainInputMode >(
 		inputMode.domainInput
 	);
 
+	// Sync site ID from stepper context to Redux store
+	// This ensures components using Redux connect() can access the selected site
+	useEffect( () => {
+		if ( siteId ) {
+			reduxDispatch( setSelectedSiteId( siteId ) );
+		}
+	}, [ siteId, reduxDispatch ] );
```

Comment:
It's weird, [we're already doing that in the flow setup if the user enters `/setup/domain` directly instead of `/setup/domain/use-my-domain`.](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/flows/domain/domain.ts#L388)

Maybe we can modify the condition in the code that I linked so it always set the site id from the hook?